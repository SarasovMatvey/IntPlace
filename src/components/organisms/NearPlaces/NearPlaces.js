import { useEffect, useState } from "react";
import { Button, Segment } from "semantic-ui-react";
import Masonry from "react-responsive-masonry";
import sygicAxios from "../../../api/sygic";
import PlaceCard from "../../atoms/PlaceCard";
import { useHistory } from "react-router";
import "./NearPlaces.sass";

function NearPlaces({ className = "", targetPlaceId, area = {} }) {
  const { lat, lon, radius } = area;

  const history = useHistory();
  const [isOpen, setIsOpen] = useState(false);
  const [places, setPlaces] = useState([]);

  useEffect(() => {
    sygicAxios(`/places/list?area=${lat},${lon},${radius}`).then(
      ({ data: { data } }) => {
        const placesWithoutSimilar = data.places.filter(
          placeInfo => !(placeInfo.id === targetPlaceId)
        );

        setPlaces(placesWithoutSimilar);
      }
    );
  }, [lat, lon, radius, targetPlaceId]);

  return (
    <div className={`near-places ${className}`}>
      <Button onClick={() => setIsOpen(isOpen => !isOpen)}>
        {isOpen ? "Hide" : "Show"} nearest places
      </Button>
      {isOpen ? (
        <Segment>
          <Masonry gutter="2rem" columnsCount={4}>
            {places.map(placeInfo => (
              <PlaceCard
                className="places-finder__card"
                key={placeInfo.id}
                fluid
                title={placeInfo.name}
                poster={placeInfo.thumbnail_url}
                description={placeInfo.perex}
                placeId={placeInfo.id}
                onClick={e => goToPlaceInfoPage(e.currentTarget.dataset.id)}
              />
            ))}
          </Masonry>
        </Segment>
      ) : null}
    </div>
  );

  function goToPlaceInfoPage(placeId) {
    history.replace(`/place/${placeId}`);
  }
}

export default NearPlaces;
