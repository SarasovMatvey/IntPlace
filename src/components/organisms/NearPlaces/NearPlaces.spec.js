import NearPlaces from "./NearPlaces";

const setUp = props => shallow(<NearPlaces {...props} />);

describe("NearPlaces", () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it("should renders properly", () => {
    expect(component).toMatchSnapshot();
  });
});
