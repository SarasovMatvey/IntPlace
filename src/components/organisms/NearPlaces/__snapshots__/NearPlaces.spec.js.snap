// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NearPlaces should renders properly 1`] = `
<div
  className="near-places "
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Show
     nearest places
  </Button>
</div>
`;
